scipy/_lib/tests/test_public_api.py

```diff
@@ -243,14 +244,16 @@ def ignore_errors(name):
 
     modnames = []
 
-    for _, modname, _ in pkgutil.walk_packages(path=scipy.__path__,
-                                               prefix=scipy.__name__ + '.',
-                                               onerror=ignore_errors):
-        if is_unexpected(modname) and modname not in SKIP_LIST:
-            # We have a name that is new.  If that's on purpose, add it to
-            # PUBLIC_MODULES.  We don't expect to have to add anything to
-            # PRIVATE_BUT_PRESENT_MODULES.  Use an underscore in the name!
-            modnames.append(modname)
+    with np.testing.suppress_warnings() as sup:
```

Comment:
You had to suppress warnings here because `scipy.misc`, `scipy.misc.common`, and/or `scipy.misc.doccer` emit `DeprecationWarning`s when they are imported directly.

IIUC, *all* of these namespaces are deprecated and slated from removal in 2.0. Should they all emit `DeprecationWarning`s?

Reply:
You mean leave the file empty. That would be a lot simpler. On the other hand people the changed behaviour would be silent.

Reply:
No, the opposite - I suggested that we raise the warning in the file; I'm really wondering if *all* of these should emit `DeprecationWarnings` if we're actually concerned that removing the files - even if there's no content - will break code.
I'll give it another day for feedback from others, but then maybe I'll merge and open an issue about it.